Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -1234,16 +1234,23 @@ namespace Babylon
             xr::Ray GetNativeRay()
             {
                 xr::Ray nativeRay{{0, 0, 0}, {0, 0, -1}};
+
                 auto originObject = m_origin.Value();
-                nativeRay.Origin.X = originObject.Get("x").ToNumber().FloatValue();
-                nativeRay.Origin.Y = originObject.Get("y").ToNumber().FloatValue();
-                nativeRay.Origin.Z = originObject.Get("z").ToNumber().FloatValue();
+                if (originObject.Has("x"))
```

Comment:
I'm assuming it's alright to check just x, but I can individually check each value or check all 3 beforehand.

Reply:
https://github.com/BabylonJS/Babylon.js/blob/master/src/LibDeclarations/webxr.d.ts#L280-L285

```typescript
interface DOMPointInit {
    w?: number;
    x?: number;
    y?: number;
    z?: number;
}
```

It looks like from the type declarations that any of the x, y, z, w can be not set. It probably makes more sense to check each one individually. Although maybe it's better to do this in the constructor?